photon-core/src/main/java/org/photonvision/vision/frame/consumer/MJPGFrameConsumer.java

```diff
@@ -174,17 +237,22 @@ public MJPGFrameConsumer(String name, int port) {
     }
 
     public void accept(CVMat image) {
+        long now = MathUtils.wpiNanoTime();
+
         if (image != null && !image.getMat().empty()) {
-            long now = MathUtils.wpiNanoTime();
             if (now - lastFrameTimeNs > MAX_FRAME_PERIOD_NS) {
                 lastFrameTimeNs = now;
                 cvSource.putFrame(image.getMat());
             }
-
-            // Make sure our disabled framerate limiting doesn't get confused
-            isDisabled = false;
-            lastEmptyTimeNs = 0;
+        } else {
+            if (now - lastFrameTimeNs > MAX_FRAME_PERIOD_NS) {
+                lastFrameTimeNs = now;
+                cvSource.putFrame(LOST_MAT);
+            }
         }
+        // Make sure our disabled framerate limiting doesn't get confused
```

Comment:
Is this comment still accurate? looks like it runs regardless of whether the frame came in good or not